src/dex/infusion/infusion.ts

```diff
@@ -0,0 +1,695 @@
+    if (!(pair && pair.pool && pair.exchange)) return null;
+    const pairState = pair?.pool.getState(blockNumber);
+
+    if (!pairState) {
+      this.logger.error(
+        `Error_orderPairParams expected reserves, got none (maybe the pool doesn't exist) ${
+          from.symbol || from.address
+        } ${to.symbol || to.address}`,
+      );
+      return null;
+    }
+
+    const fee = (pairState.feeCode + tokenDexTransferFee).toString();
+    const pairReversed =
+      pair.token1.address.toLowerCase() === from.address.toLowerCase();
+    if (pairReversed) {
+      return {
+        tokenIn: from.address,
+        tokenOut: to.address,
+        reservesIn: pairState.reserves1,
+        reservesOut: pairState.reserves0,
+        fee,
+        direction: false,
+        exchange: pair.exchange,
+        decimalsIn: from.decimals,
+        decimalsOut: to.decimals,
+        stable,
+      };
+    }
+    return {
+      tokenIn: from.address,
+      tokenOut: to.address,
+      reservesIn: pairState.reserves0,
+      reservesOut: pairState.reserves1,
+      fee,
+      direction: true,
+      exchange: pair.exchange,
+      decimalsIn: from.decimals,
+      decimalsOut: to.decimals,
+      stable,
+    };
+  }
+
+  async getPoolIdentifiers(
+    _from: Token,
+    _to: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (side === SwapSide.BUY) return [];
+
+    const from = this.dexHelper.config.wrapETH(_from);
+    const to = this.dexHelper.config.wrapETH(_to);
+
+    if (from.address.toLowerCase() === to.address.toLowerCase()) {
+      return [];
+    }
+
+    const tokenAddress = [from.address.toLowerCase(), to.address.toLowerCase()]
+      .sort((a, b) => (a > b ? 1 : -1))
+      .join('_');
+
+    const poolIdentifier = `${this.dexKey}_${tokenAddress}`;
+    const poolIdentifierUniswap = poolIdentifier + this.poolPostfix(false);
+    const poolIdentifierStable = poolIdentifier + this.poolPostfix(true);
+    return [poolIdentifierUniswap, poolIdentifierStable];
+  }
+
+  poolPostfix(stable: boolean) {
+    return stable ? 'S' : 'V';
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  getAdapterParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    toAmount: NumberAsString, // required for buy case
+    data: InfusionData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      targetExchange: data.router,
+      payload: '',
+      networkFee: '0',
+    };
+  }
+
+  getDexParam(
+    src: Address,
+    dest: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: InfusionData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    if (side === SwapSide.BUY) throw new Error(`Buy not supported`);
+
+    let routerMethod: any;
+    let routerArgs: InfusionParam;
+    const stable = isStablePair(this.network, src, dest);
```

Comment:
Can we somehow add an `isStable` parameter to `InfusionData` so that we don't need to use a this util function with hardcoded list of stables?

Reply:
In any case we'd need to call this function, as I can see there's finite number of tokens considered as stables, so seems ok to leave as is

Reply:
Not quite sure why then `InfusionData` was updated. There may be other pools with other “stable” tokens that may not be on the list, or pools that have tokens from this hard-coded list but this pool is not "stable"

Reply:
as I can see it's just for them to know for encoding, whether it's stable, so it's needed in the encoding, also I updated InfusionData to just be UniswapV2 data, as they don't use new param